services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_api.py

```diff
@@ -29,6 +31,53 @@
 log = logging.getLogger(__name__)
 
 
+def _monkey_patch_aiodocker() -> None:
+    """Raises an error once the library is up to date."""
+    from distutils.version import LooseVersion
+
+    from aiodocker import volumes
+    from aiodocker.utils import clean_filters
+    from aiodocker.volumes import DockerVolume
+
+    if LooseVersion(aiodocker.__version__) > LooseVersion("0.21.0"):
+        raise RuntimeError(
+            "Please check that PR https://github.com/aio-libs/aiodocker/pull/623 "
+            "is not part of the current bump version. "
+            "Otherwise, if the current PR is part of this new release "
+            "remove monkey_patch."
+        )
+
+    # pylint: disable=protected-access
+    # pylint: disable=redefined-builtin
+    async def list(self, *, filters=None):
+        """
+        Return a list of volumes
+
+        Args:
+            filters: a dict with a list of filters
+
+        Available filters:
+            dangling=<boolean>
+            driver=<volume-driver-name>
+            label=<key> or label=<key>:<value>
+            name=<volume-name>
+        """
+        params = {} if filters is None else {"filters": clean_filters(filters)}
+
+        data = await self.docker._query_json("volumes", params=params)
+        return data
+
+    async def get(self, id):
+        data = await self.docker._query_json("volumes/{id}".format(id=id), method="GET")
+        return DockerVolume(self.docker, data["Name"])
+
+    volumes.DockerVolumes.list = list
+    setattr(volumes.DockerVolumes, "get", get)
+
```

Comment:
why ``get`` and ``list`` are overriden differently? 

See ``types.MethodType``. At first sight (double check pls), this is how I would do it
```python
import types

volumes.DockerVolumes.list = types.MethodType(_custom_volumes_list, app)
volumes.DockerVolumes.get= types.MethodType(_custom_volumes_get, app)
```

Reply:
Because list already exists and get does not exist. Will use setattr for both.